sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkGetColumnsOperation.scala

```diff
@@ -171,8 +171,8 @@ private[hive] class SparkGetColumnsOperation(
     case BooleanType => java.sql.Types.BOOLEAN
     case ByteType => java.sql.Types.TINYINT
     case ShortType => java.sql.Types.SMALLINT
-    case IntegerType => java.sql.Types.INTEGER
-    case LongType => java.sql.Types.BIGINT
+    case IntegerType | YearMonthIntervalType => java.sql.Types.INTEGER
+    case LongType | DayTimeIntervalType => java.sql.Types.BIGINT
```

Comment:
hmm, I am not sure that we should expose ANSI intervals as raw integers/longs via JDBC. I would consider strings (preferable) or java.time.Duration/Period. @cloud-fan @srielau WDYT?

Reply:
Good question! I have the confusion too.

Reply:
This is the metadata, where do we handle the data? e.g. if we want to return string or Duration/Period, where shall we instantiate string or Duration/Period values?

Reply:
See the PR https://github.com/apache/spark/pull/32121. We should return strings. Since there is no appropriate type in `java.sql.*` for intervals. I believe we should return the same as for `CalendarIntervalType` - `java.sql.Types.OTHER`.

@beliefer Could you handle `YearMonthIntervalType` and `DayTimeIntervalType` separately, and return `java.sql.Types.OTHER`

Reply:
OK